component/authenticator/src/main/java/org/wso2/carbon/identity/authenticator/emailotp/EmailOTPAuthenticator.java

```diff
@@ -330,6 +331,21 @@ protected void processAuthenticationResponse(HttpServletRequest request, HttpSer
             succeededAttempt = validateWithBackUpCodes(context, userToken, authenticatedUser);
         }
 
+        if (succeededAttempt && isLocalUser) {
+            String username = authenticatedUser.toFullQualifiedUsername();
+            try {
+                String userEmail = getEmailValueForUsername(username, context);
+                if (StringUtils.isEmpty(userEmail)) {
```

Comment:
let's use StringUtils.isBlank here

Reply:
also shall we move the userEmail validation logic inside updateEmailAddressForUsername method?

Reply:
First issue fixed. Second issue could you please explain it a bit.